### DISCUSSION TITLE
How to properly add XML support

### DISCUSSION THREAD

**Discussion by rabestro** [usebruno/bruno#1335] [Q&A] [answered]
I need to validate XML responses with Bruno CLI. I've added the dependencies:

```json
    "dependencies": {
        "@usebruno/cli": "^1.3.0",
        "xml2js": "^0.6.2"
    }
```

When I run test:
```
tests {
  const { xml2js } = require('xml2js');

  test("should return error message", function() {
    xml2js.parseString(res.getBody(), function (err, result) {
      expect(err).to.be.null;
      expect(result.TranslationResponse.sentence[0]).to.equal('ellowyay ubmarinesay');
    });
  });
}
```

I've got a message: Cannot find module 'events'

Full message: 
Error invoking remote method 'send-http-request': VMError: Cannot find module 'events'

The GitHub repository:
https://github.com/rabestro/pig-latin-rest

Failing run:
https://github.com/rabestro/pig-latin-rest/runs/20211225193

**Comment by helloanoop** [comment, marked as answer]
@rabestro 

You need to whitelist the nodejs `events` module.
Can you add this and try in your package.json

`bruno.json`
```json
{
  "scripts": {
    "moduleWhitelist": [
      "events"
    ]
  }
}
```

**Reply by rabestro** [reply]
Thank you for a prompt answer! 

I've added this and one more module to the script section of `bruno.json` file. 

```json
{
    "version": "1",
    "name": "Pig Latin translator",
    "type": "collection",
    "presets": {
        "requestType": "http",
        "requestUrl": ""
    },
    "scripts": {
        "moduleWhitelist": [
            "events", "timers"
        ]
    }
}
```

Now this test works fine:

```javascripe
  const xml2js = require('xml2js');

  test("should translate message", function() {
    xml2js.parseString(res.getBody(), function (err, result) {
      expect(err).to.be.null;
      const actualSentence = result.TranslationResponse.sentence[0].trim()
      expect(actualSentence).to.equal('ellowyay ubmarinesay');
    });
  });
```

The test report is available here:
https://github.com/rabestro/pig-latin-rest/runs/20226322262
